finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/ServerBridge.scala

```diff
@@ -15,6 +17,17 @@ private[netty4] class ServerBridge[In, Out](
 
   def initChannel(ch: Channel): Unit = {
     val transport: Transport[In, Out] = transportFac(ch)
-    serveTransport(transport)
```

Comment:
How do you know it never happens? This looks pretty reasonable to me. As far as I remember, the handshake is initiated on either `channelActive` or `handlerAdded`. Here you're attaching the handshake listener on `channelRegistered`, but it shouldn't matter and the listener should be fired when the handshake is done anyway.

Maybe just make it a pure ChannelHandler and override (and delay) `channelActive` instead?